### ISSUE TITLE
BUG: URI "no file associated with this alert" in SARIF now invalid in github/codeql-action

### ISSUE THREAD

**Issue by michaelkedar** [ossf/scorecard#3063]
**Describe the bug**
Currently, the URI in SARIF files for alerts without locations is [set to the string "no file associated with this alert" ](https://github.com/ossf/scorecard/blob/68c23e166d9eda1a0d66653421169137b51dbe31/pkg/sarif.go#L694-L696C36).
Using [`github/codeql-action/upload-sarif@v2.3.4`](https://github.com/github/codeql-action/tree/f0e3dfb30302f8a0881bb509b044e0de4f6ef589), this causes errors:
```
Error details: instance.runs[0].results[0].locations[0].physicalLocation.artifactLocation.uri does not conform to the "uri-reference" format
Error: Unable to upload "results.sarif" as it is not valid SARIF
```

**Reproduction steps**
1. Have scorecard generate a SARIF file with an alert with no associated file
2. Pass SARIF file to `github/codeql-action/upload-sarif@v2.3.4` action

**Expected behavior**
This shouldn't cause an error.

**Additional context**
The schema file was updated here:
https://github.com/github/codeql-action/commit/febbadf751ad5dff69b0db9d9f7ff0824e228e05

**Comment by laurentsimon** [comment]
Created https://github.com/github/codeql-action/issues/1703 on their side. They made a breaking change without following semver. AFAIR, the URI needs to be populated otherwise the results don't show up for results that don't have a "path". 

We can to try to generate a conformant URI that does not correspond to a file in the repo...

**Comment by spencerschrock** [comment]
This is resolved by upgrading to `github/codeql-action` v2.3.5.
